src/wallet/wallet.cpp

```diff
@@ -3140,7 +3140,9 @@ bool CWallet::InitLoadWallet()
     RegisterValidationInterface(walletInstance);
 
     CBlockIndex *pindexRescan = chainActive.Tip();
-    if (GetBoolArg("-rescan", false))
+    if (GetBoolArg("-rescan", false) && GetArg("-rescan", 0) > 0)
+        pindexRescan = chainActive[GetArg("-rescan", 0)];
+    else if (GetBoolArg("-rescan", false))
```

Comment:
This doesn't need to be a special case since the default for `-rescan` is `0`, and `chainActive.Genesis() == chainActive[0]`